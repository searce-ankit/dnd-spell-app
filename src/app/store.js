import { configureStore } from '@reduxjs/toolkit';
import favouriteSpellReducer from '../features/spell/spellSlice';

export const store = configureStore({
  reducer: {
    favouriteSpell: favouriteSpellReducer,
  },
});
