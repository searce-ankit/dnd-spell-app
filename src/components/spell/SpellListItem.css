.spell-li{
    padding: 10px;
    text-align: left;
    border-bottom: solid 1px #ccc;
    cursor: pointer;
}
.spell-li:hover{
    background-color: #efefef;
}

.spell-li > a{
    text-decoration: none;
}

.spell-li > .fav-icon{
    float: right;
}