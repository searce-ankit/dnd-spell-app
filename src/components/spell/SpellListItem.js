import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faBookmark } from '@fortawesome/free-solid-svg-icons';
import { useDispatch } from 'react-redux';
import { addToFavourite,removeFromfavourite } from '../../features/spell/spellSlice';

import './SpellListItem.css';

export function SpellListItem(props){

    const dispatch = useDispatch();
    let spell=props.spell;
    let isFavourite=props.isFavourite;

    const bookMarkHandler = () => {
        if(isFavourite){
            dispatch(removeFromfavourite(spell.index));
        }else{
            dispatch(addToFavourite(spell));
            if(props.updateSpells){
                props.updateSpells();
            }
        }
    }

    return (<div className='spell-li'>
       <Link to={"/spell/"+ spell.index}>{ spell.name }</Link>
       <FontAwesomeIcon className='fav-icon' icon={faBookmark} color={isFavourite?"gold":"gray"} onClick={bookMarkHandler} />
    </div>);
}