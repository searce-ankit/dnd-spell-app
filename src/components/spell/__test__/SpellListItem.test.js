import React from "react";
import { render,cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../app/store';
import { SpellListItem } from "../SpellListItem";
import { BrowserRouter as Router,Routes,Route } from 'react-router-dom';
import renderer  from 'react-test-renderer';


afterEach(cleanup);

it('renders spell list item', () => {
    const spell={
        index:'expecto-patronum',
        name:'expecto patronum'
    };
    const { getByText } = render(
        <Provider store={store}>
            <Router>
                <SpellListItem key={spell.index} spell={spell} isFavourite={ false } />
            </Router>
        </Provider>
    );
  
    expect(getByText(/expecto patronum/i)).toBeInTheDocument();
});

it('maches snapshot',()=>{

    const spell={
        index:'expecto-patronum',
        name:'expecto patronum'
    };

    const tree = renderer.create(<Provider store={store}>
        <Router>
            <SpellListItem key={spell.index} spell={spell} isFavourite={ false } />
        </Router>
    </Provider>).toJSON();
    
    expect(tree).toMatchSnapshot();

});