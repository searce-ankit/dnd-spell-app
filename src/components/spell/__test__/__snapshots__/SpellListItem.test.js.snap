// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maches snapshot 1`] = `
<div
  className="spell-li"
>
  <a
    href="/spell/expecto-patronum"
    onClick={[Function]}
  >
    expecto patronum
  </a>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-bookmark fav-icon"
    color="gray"
    data-icon="bookmark"
    data-prefix="fas"
    focusable="false"
    onClick={[Function]}
    role="img"
    style={Object {}}
    viewBox="0 0 384 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M384 48V512l-192-112L0 512V48C0 21.5 21.5 0 48 0h288C362.5 0 384 21.5 384 48z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</div>
`;
